fit-checker/utils/change-detect/ClassificationDetect.swift

```diff
@@ -0,0 +1,162 @@
+                    -> [Change<ClassificationRow>] {
+        let oldTables = oldValue.tables
+        let newTables = newValue.tables
+        var changes = [Change<ClassificationRow>]()
+
+        var oldIndex = 0
+        var newIndex = 0
+
+        while oldIndex < oldTables.count || newIndex < newTables.count {
+            let oldTable = ArrayUtils<ClassificationTable>.getItemSafely(array: oldTables, i: oldIndex)
+            let newTable = ArrayUtils<ClassificationTable>.getItemSafely(array: newTables, i: newIndex)
+
+            let tableChanges = detectChangeForTable(oldTable: oldTable, newTable: newTable)
+
+            changes.append(contentsOf: tableChanges)
+
+            oldIndex += 1
+            newIndex += 1
+        }
+
+        return changes
+    }
+
+
+    /// Iterates over rows in tables and detect changes there.
+    ///
+    /// - Parameters:
+    ///   - oldTable: ClassificationTable?
+    ///   - newTable: ClassificationTable?
+    /// - Returns: changes
+    private func detectChangeForTable(oldTable: ClassificationTable?, newTable: ClassificationTable?)
+                    -> [Change<ClassificationRow>] {
+        let oldRows = getRowsSafely(table: oldTable)
+        let newRows = getRowsSafely(table: newTable)
+        var tableChanges = [Change<ClassificationRow>]()
+
+        var oldRowIndex = 0
+        var newRowIndex = 0
+
+        while oldRowIndex < oldRows.count || newRowIndex < newRows.count {
+            let oldRow = ArrayUtils<ClassificationRow>.getItemSafely(array: oldRows, i: oldRowIndex)
+            let newRow = ArrayUtils<ClassificationRow>.getItemSafely(array: newRows, i: newRowIndex)
+
+            let change = detectValueChange(oldItem: oldRow, newItem: newRow)
+            if let change = change {
+                tableChanges.append(change)
+            }
+
+            oldRowIndex += 1
+            newRowIndex += 1
+        }
+
+        return tableChanges
+    }
+
+
+    /// Gets rows from table safely.
+    ///
+    /// - Parameter table: table with rows
+    /// - Returns: rows or empty array
+    private func getRowsSafely(table: ClassificationTable?) -> [ClassificationRow] {
+        if let table = table {
+            return table.rows
+        } else {
+            return [ClassificationRow]()
+        }
+    }
+
+
+    /// Detects change in row.
+    ///
+    /// - Parameters:
+    ///   - oldItem: ClassificationRow
+    ///   - newItem: ClassificationRow
+    /// - Returns: found change or empty
+    private func detectValueChange(oldItem: ClassificationRow?, newItem: ClassificationRow?)
+                    -> Change<ClassificationRow>? {
+        var type: ChangeType?
+
+        type = (oldItem == nil ? ChangeType.added : nil)
+
+        type = (newItem == nil ? ChangeType.removed : type)
+
+        if let old = oldItem, let new = newItem {
+            if (old.name != new.name || old.value != new.value) {
```

Comment:
Could we implemented operator `==` for ClassificationRow? I think it's more readable and also more DRY than plain properties comparsion..